### PR TITLE
431710 report something

### PR DESCRIPTION
**Pull request by jbarnsley10**

### CODE REVIEW

File: app/api/ddi-index-api/dog-breaches.js [hunk 1]
```diff
@@ -9,21 +9,32 @@ const dogBreachesEndpoint = 'breaches'
...
  */
 const getBreachCategories = async (user) => {
   const payload = await get(`${dogBreachesEndpoint}/categories`, user)
 
-  return payload.breachCategories.map(breachCategory => {
-    const [firstLetter, ...restOfLetters] = breachCategory.label
-    const label = firstLetter.toUpperCase() + restOfLetters.join('')
+  return payload.breachCategories
+    .filter(cat => categoriesShortNamesForDisplay.includes(cat.short_name))
+    .map(breachCategory => {
```
Comment (whitewaterdesign) [inline app/api/ddi-index-api/dog-breaches.js:29]: This works well. 👍

File: app/api/os-places.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+const postcodeEndpoint = 'postcode'
+
+const options = {
+  headers: {
+    key: config.osPlacesApi.token
+  },
+  json: true
+}
+
+const getPostcodeAddresses = async (postcode, houseNumber) => {
```
Comment (whitewaterdesign) [inline app/api/os-places.js:15]: Would it be too much to add jsdoc to this, would be helpful for future, especially with the nested arrays

File: app/constants/forms.js [hunk 3]
```diff
@@ -1,9 +1,11 @@
 const constants = {
   errorCodes: {
-    validationError: 400
+    validationError: 400,
+    notFoundError: 404
   },
```
Comment (whitewaterdesign) [inline app/constants/forms.js:5]: Like this 👍 would be good to do in portal too when we next get a chance.

Comment (jbarnsley10) [reply app/constants/forms.js:5]: Haha - I didn't have a choice as SonarCloud thought it was a critical error to use an explicit 400 or 404 etc without having them in a constant. It says 'No magic number'. Seems to be a new thing that it reports

File: app/lib/report-helper.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  } else if (selected === reportTypes.dogDied) {
+    return `${routes.dogDied.get}${backNav?.srcHashParam}`
+  } else {
+    return `${routes.somethingElse.get}${backNav?.srcHashParam}`
+  }
+}
+
+const sendReportEmail = async (data) => {
+  console.log('JB send email', data)
+}
```
Comment (whitewaterdesign) [inline app/lib/report-helper.js:60]: can we change the log text here?

File: app/routes/cdo/report/something-else.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  },
+  {
+    method: 'POST',
+    path: routes.somethingElse.post,
+    options: {
+      validate: {
+        payload: validatePayload,
+        failAction: async (request, h, error) => {
+          console.log('POST something-else validation error', error)
+          const payload = { ...getReportTypeDetails(request), ...request.payload }
```
Comment (whitewaterdesign) [inline app/routes/cdo/report/something-else.js:44]: stray console log?

File: docker-compose.yaml [hunk 6]
```diff
@@ -10,6 +10,8 @@ services:
       APPLICATIONINSIGHTS_CONNECTION_STRING: ${APPLICATIONINSIGHTS_CONNECTION_STRING}
       DDI_API_BASE_URL: ${DDI_API_BASE_URL:-test}
       DDI_EVENTS_BASE_URL: ${DDI_EVENTS_BASE_URL:-test}
+      OS_PLACES_API_KEY: ${OS_PLACES_API_KEY:-test}
+      OS_PLACES_API_BASE_URL: ${OS_PLACES_API_BASE_URL:-test}
       COOKIE_PASSWORD: who-likes-cookies-i-like-cookies-everybody-likes-cookies
```
Comment (whitewaterdesign) [inline docker-compose.yaml:15]: Do we need this in docker-compose.test as well?

Comment (whitewaterdesign) [reply docker-compose.yaml:15]: can the defaults reflect the ones in jest.setup.single?  if they are different then the tests error when you running them locally?

**Review by whitewaterdesign — APPROVED** [review]
Looks good

**Pull request merged**